src/rgw/rgw_rest_s3.cc

```diff
@@ -6356,6 +6356,7 @@ rgw::auth::s3::LocalEngine::authenticate(
       throw -EPERM;
     }
   }*/
+  s->user->set_attrs(user->get_attrs());
```

Comment:
hmm this call to `authenticate()` may end up failing, and a different auth engine could succeed with a different identity. this would use the wrong user's attrs for the request

Reply:
@cbodley ack, thank you
pushed an update which moves the `s->user->set_attrs(...)` to a positions above a `return result_t::grant(...)`, at those positions checks have completed.

yet, two cases that not sure about are in:
https://github.com/ceph/ceph/blob/ee6523b618e7abe21cd2346f828f57cf150de5b9/src/rgw/rgw_rest_s3.cc#L6593
are `return result_t::grant(...)` from 
https://github.com/ceph/ceph/blob/ee6523b618e7abe21cd2346f828f57cf150de5b9/src/rgw/rgw_rest_s3.cc#L6691
and
https://github.com/ceph/ceph/blob/ee6523b618e7abe21cd2346f828f57cf150de5b9/src/rgw/rgw_rest_s3.cc#L6695
should also `s->user->set_attrs(...)` ?